core/authenticate/service.go

```diff
@@ -316,6 +385,208 @@ func (s Service) applyMailOTP(ctx context.Context, request RegistrationFinishReq
 	}, nil
 }
 
+func (s Service) startPassKeyRegisterMethod(ctx context.Context, flow *Flow) (*RegistrationStartResponse, error) {
+	newPassKeyUser := strategy.NewPassKeyUser(flow.Email)
+	options, session, err := s.webAuth.BeginRegistration(newPassKeyUser)
+	if err != nil {
+		return nil, err
+	}
+
+	session.Challenge = base64.RawURLEncoding.EncodeToString([]byte(session.Challenge))
+	sessionInBytes, err := json.Marshal(session)
+	if err != nil {
+		return nil, err
+	}
+	flow.Metadata["passkey_session"] = sessionInBytes
+	flow.Metadata["passkey_type"] = "register"
+	if err = s.flowRepo.Set(ctx, flow); err != nil {
+		return nil, err
+	}
+	optionsBytes, err := json.Marshal(options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RegistrationStartResponse{
+		Flow:  flow,
+		State: flow.ID.String(),
+		StateConfig: map[string]any{
+			"options": optionsBytes,
+		},
+	}, nil
+}
+
+func (s Service) startPassKeyLoginMethod(ctx context.Context, loggedInUser user.User, flow *Flow) (*RegistrationStartResponse, error) {
+	newPassKeyUser := strategy.NewPassKeyUser(flow.Email)
```

Comment:
We can also create a constructer that accepts credentials like `strategy.NewPasskeyUserWithCredentials(flow.Email, webAuthCredentialData)` instead of doing `newPassKeyUser.Credentials = webAuthCredentialData`

Reply:
done